plugins/woocommerce/tests/php/includes/wc-core-functions-test.php

```diff
@@ -43,4 +43,85 @@ public function test_wc_asort_by_locale() {
 
 		$this->assertSame( array( 'Afghanistan', 'Åland Islands', 'Espagne', 'Éthiopie' ), array_values( $sorted_values ) );
 	}
+
+	/**
+	 * @testdDox wc_get_rounding_precision returns the value of wc_get_price_decimals()+2, but with a minimum of WC_ROUNDING_PRECISION (6)
+	 *
+	 * @testWith [0, 6]
+	 *           [2, 6]
+	 *           [4, 6]
+	 *           [5, 7]
+	 *           [6, 8]
+	 *           [7, 9]
+	 *
+	 * @param int $decimals Value returned by wc_get_price_decimals().
+	 * @param int $expected Expected value returned by the function.
+	 */
+	public function test_wc_get_rounding_precision( $decimals, $expected ) {
+		add_filter(
+			'wc_get_price_decimals',
+			function() use ( $decimals ) {
+				return $decimals;
+			}
+		);
+
+		$actual = wc_get_rounding_precision();
+		$this->assertEquals( $expected, $actual );
+
+		remove_all_filters( 'wc_get_price_decimals' );
+	}
+
+	/**
+	 * @testDox wc_add_number_precision moves the decimal point to the right as many places as wc_get_price_decimals() says, and (optionally) properly rounds the result.
+	 *
+	 * @testWith [2, 1.23456789, false, 123.456789]
```

Comment:
TIL ... I didn't realize this was possible (thought we needed a data provider method).

Reply:
For small and scalar-only inputs it's a very practical and (my favorite thing) readable alternative to data providers.

Reply:
Yep, that's a big readability win ❤️